crates/core_simd/src/elements/const_ptr.rs

```diff
@@ -0,0 +1,66 @@
+use super::sealed::Sealed;
+use crate::simd::{LaneCount, Mask, Simd, SimdPartialEq, SupportedLaneCount};
+
+/// Operations on SIMD vectors of constant pointers.
+pub trait SimdConstPtr: Copy + Sealed {
+    /// Vector of usize with the same number of lanes.
+    type Usize;
+
+    /// Vector of mutable pointers to the same type.
+    type MutPtr;
+
+    /// Mask type used for manipulating this SIMD vector type.
+    type Mask;
+
+    /// Returns `true` for each lane that is null.
+    fn is_null(self) -> Self::Mask;
+
+    /// Changes constness without changing the type.
+    fn as_mut(self) -> Self::MutPtr;
+
+    /// Gets the "address" portion of the pointer.
+    ///
+    /// Equivalent to calling [`pointer::addr`] on each lane.
+    fn addr(self) -> Self::Usize;
```

Comment:
the docs need some elaboration -- especially specifically mentioning that the result can't be cast back to a pointer and used.
https://github.com/rust-lang/rust/blob/10f4ce324baf7cfb7ce2b2096662b82b79204944/library/core/src/ptr/const_ptr.rs#L178

Reply:
Thought it would be better to reference the underlying function it's equivalent to, especially considering it's likely to change sometime before stabilization

Reply:
yes, but imho we need to point out that you can't use it like `ptr as usize` which allows getting `ptr` back by `as` casting the resulting integer back to a pointer. otherwise people who only read the docs of this function may assume it's `ptr as usize`, leading them to write code with UB.